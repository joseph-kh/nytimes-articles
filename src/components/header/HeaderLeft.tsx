import React from "react";
import COLORS from "@/theme/colors";
import { MaterialIcons } from "@expo/vector-icons";
import { TouchableOpacity } from "react-native-gesture-handler";
import { moderateScale } from "@/utils/scaler";

const HeaderLeft: React.FC = () => {
  return (
    <TouchableOpacity>
      <MaterialIcons
        name="menu"
        size={moderateScale(24)}
        color={COLORS.white}
      />
    </TouchableOpacity>
  );
};

export default HeaderLeft;
