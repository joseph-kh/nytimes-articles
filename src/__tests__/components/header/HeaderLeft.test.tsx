import React from "react";
import renderer from "react-test-renderer";
import HeaderLeft from "@/components/header/HeaderLeft";

describe("HeaderLeft Component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<HeaderLeft />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
