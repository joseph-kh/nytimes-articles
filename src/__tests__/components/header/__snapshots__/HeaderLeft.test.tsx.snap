// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLeft Component renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</RNGestureHandlerButton>
`;
